.github/workflows/build-test-vm.yaml

```diff
@@ -86,9 +94,55 @@ jobs:
           username: ${{ secrets.NEON_DOCKERHUB_USERNAME }}
           password: ${{ secrets.NEON_DOCKERHUB_PASSWORD }}
 
-      - name: build ${{ needs.tags.outputs.vm-postgres-16-bullseye }}
+      - name: build ${{ steps.tags.outputs.vm-postgres-16-bullseye  }}
         run: |
-          ./bin/vm-builder -src postgres:16-bullseye -spec tests/e2e/image-spec.yaml -dst ${{ needs.tags.outputs.vm-postgres-16-bullseye }} -daemon-image ${{ needs.tags.outputs.daemon }} -target-arch linux/${TARGET_ARCH}
-      - name: docker push ${{ needs.tags.outputs.vm-postgres-16-bullseye }}
+          ./bin/vm-builder -src postgres:16-bullseye -spec tests/e2e/image-spec.yaml -dst ${{ steps.tags.outputs.vm-postgres-16-bullseye  }} -daemon-image ${{ steps.tags.outputs.daemon }} -target-arch linux/${{ matrix.arch }}
+
+      # vm-builder doesn't attach platform annotation to the image, so we need to do it manually
+      - name: annotate image with platform and push
         run: |
-          docker push ${{ needs.tags.outputs.vm-postgres-16-bullseye }}
+          # docker manifest command works only with remote images so we need to push the image first
+          docker push ${{ steps.tags.outputs.vm-postgres-16-bullseye }}
+          docker buildx imagetools inspect ${{ steps.tags.outputs.vm-postgres-16-bullseye }} --format '{{ .Manifests }}'
+          docker manifest create ${{ steps.tags.outputs.vm-postgres-16-bullseye }} ${{ steps.tags.outputs.vm-postgres-16-bullseye }}
+          docker manifest annotate ${{ steps.tags.outputs.vm-postgres-16-bullseye }} ${{ steps.tags.outputs.vm-postgres-16-bullseye }} --os linux --arch ${{ matrix.arch }}
+          docker manifest push ${{ steps.tags.outputs.vm-postgres-16-bullseye }}
+
+
+  merge-vm-images:
+    needs: [build]
+    runs-on: ubuntu-latest
+    outputs:
+      vm-postgres-16-bullseye: ${{ steps.merge-vm-images.outputs.vm-postgres-16-bullseye }}
+    env:
+      ARCHS: ${{ join(fromJson(inputs.archs), ' ') }}
+      TAG: ${{ inputs.tag }}
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v4
+
+      - name: set custom docker config directory
+        uses: neondatabase/dev-actions/set-docker-config-dir@6094485bf440001c94a94a3f9e221e81ff6b6193
```

Comment:
question: Should we remove this? (cc @bayandin)

Saw the following warning while reviewing:

> Unnecessary usage of `.github/actions/set-docker-config-dir`
>
> No need to use `.github/actions/set-docker-config-dir` action on GitHub-hosted runners

Reply:
Yes, it should be dropped here (should be kept only for self-hosted runners if I recall correctly)